src/tests/test_csv_parser.cpp

```diff
@@ -0,0 +1,95 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "parser/csv_parser/csv_parser.h"
+#include "table_config.h"
+
+namespace tests {
+
+using ::testing::ContainerEq;
+
+class TestCSVParser : public ::testing::Test {};
+
+static void CheckGetNextRow(std::filesystem::path const& path, char separator, bool has_header,
+                            size_t num_rows, std::vector<std::vector<std::string>> expected) {
+    auto parser = std::make_unique<CSVParser>(path, separator, has_header);
+    if (has_header) num_rows--;
+
+    std::vector<std::vector<std::string>> actual;
+    std::vector<std::string> row;
+    for (size_t index = 0; index < num_rows; index++) {
+        row = parser->GetNextRow();
+        actual.push_back(row);
+    }
+
+    ASSERT_THAT(actual, ContainerEq(expected)) << "Fail on " << path.c_str();
+}
+
+TEST(TestCSVParser, TestGetNextRow) {
+    CheckGetNextRow(test_data_dir / "NullEmpty.csv", ',', true, 4,
```

Comment:
it's better to use `TableConfig` from `all_tables_config.h`.
Add new tables if necessary.